airbyte-integrations/connectors/source-sendgrid/source_sendgrid/manifest.yaml

```diff
@@ -1,257 +1,987 @@
-  - $ref: "#/definitions/base_stream"
-    $parameters:
-      name: "unsubscribe_groups"
-      primary_key: "id"
-      path: "/v3/asm/groups"
-      field_path: []
-check:
-  type: CheckStream
-  stream_names: ["lists"]
+          generations: legacy,dynamic
+        request_headers: {}
+        authenticator:
+          type: BearerAuthenticator
+          api_token: "{{ config['api_key'] }}"
+        request_body_json: {}
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - result
+      paginator:
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestPath
+        page_size_option:
+          type: RequestOption
+          field_name: page_size
+          inject_into: request_parameter
+        pagination_strategy:
+          type: CursorPagination
+          page_size: 200
+          cursor_value: >-
+            {{ response['_metadata']['next'].replace('https://api.sendgrid.com',
+            '') if response['_metadata'].get('next') else None }}
+      partition_router: []
+  - type: DeclarativeStream
+    name: campaigns
+    primary_key:
+      - id
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $schema: http://json-schema.org/schema#
+        properties:
+          channels:
+            items:
+              type: string
+            type: array
+          created_at:
+            type: string
+          id:
+            type: string
+          is_abtest:
+            type: boolean
+          name:
+            type: string
+          status:
+            type: string
+          updated_at:
+            type: string
+        type: object
+    retriever:
+      type: SimpleRetriever
+      requester:
+        type: HttpRequester
+        url_base: https://api.sendgrid.com/v3
+        path: /marketing/campaigns
+        http_method: GET
+        request_parameters: {}
+        request_headers: {}
+        authenticator:
+          type: BearerAuthenticator
+          api_token: "{{ config['api_key'] }}"
+        request_body_json: {}
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - result
+      paginator:
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestPath
+        page_size_option:
+          type: RequestOption
+          field_name: page_size
+          inject_into: request_parameter
+        pagination_strategy:
+          type: CursorPagination
+          page_size: 100
+          cursor_value: >-
+            {{ response['_metadata']['next'].replace('https://api.sendgrid.com',
+            '') if response['_metadata'].get('next') else None }}
+      partition_router: []
+spec:
+  connection_specification:
+    $schema: http://json-schema.org/draft-07/schema#
+    type: object
+    required:
+      - start_date
```

Comment:
According to the original specification, it is not a required parameter. However, its absence has caused issues with previously configured connectors.

Reply:
Yeah, I want to test without it and see what happens. Do we know if the low code framework is going to handle it being None/empty @girarda ?

Reply:
I tried without it and it crashed. Leving it in and added migration code that set it to sendgrid's creation.